### ISSUE TITLE
Unicable II (JESS) Dynamic Mode Not Working

### ISSUE THREAD

**Issue by Neolo** [catalinii/minisatip#840]
Minisatip has some weird approach when communicating with Unicable II switches, which are configured in DYNAMIC mode. In fact, it does tune NOTHING over configured slot and freq, unless the same slot was used in other tuner prior to minisatip. So I have to tune the slot in Astra Cesbo or just external tuner box, and then it starts working in Minisatip. Putting or removing PIN makes no difference. 
This is non-sence, once I reboot the switch, minisatip is unable to tune anything by it's own. **### Other software like TVH or Astra or external tuners have no such issues.**

...-a 0:0:0 -j 0:13-1400,1:23-1750,2:5-1155,3:14-1435 -M *:5.0-5.0...
I have to literally "_poke_" every slot from this line in Astra Cesbo, like 23 freq 1750 etc. Only after that it tunes over minisatip. Otherwise only empty signal from the switch is displayed in Minisatip web-gui.

**Comment by Neolo** [comment]
After intensive tests, I can say for sure that JESS support is broken 100%, it just picks up the active UB slot after Astra Cesbo and continues receiving the transponder. Diseqc sequence is not being sent over sat>ip RTP or something.
In this case TVH is a sat>ip client, minisatip is a server.

**Comment by 9000h** [comment]
provide a log and tell us the type of Unicable II switch you use

**Comment by Neolo** [comment]
> provide a log and tell us the type of Unicable II switch you use

To do not post a huge unrelated log, I highlight only that part, which is repeating, simply like no signal.
Inverto IDLU-UWT110-CUO1O-32P and Terra SRQ540, same thing with both. The only acceptable solution here would be to set the slots, used with minisatip, to static mode and assign needed transponders, but this is so dull and requires switch restart every time I need to change something.

[24/04 05:39:07.024 main]: set adapter 2 for sid 0 m:0 s:1
[24/04 05:39:07.024 main]: setting DVB parameters for adapter 2 - master_sid 0 sid 0 old f:0
[24/04 05:39:07.024 main]: deleting pids on adapter 2, sid -1, pids=NULL
[24/04 05:39:07.024 main]: copy_dvb_param start -> src=0, fe=0, freq=0, fec=9, sr=0, pol=0, ro=3, msys=0, mtype=6, plts=2, bw=8000000, inv=2, pids=NULL, apids=NULL, dpids=NULL x_pmt=NULL
[24/04 05:39:07.024 main]: copy_dvb_parameters -> src=1, fe=1, freq=12399000, fec=9 sr=27500000, pol=2, ro=3, msys=6, mtype=9, plts=2, bw=8000000, inv=2, pids=0, apids=326,406,3306, dpids=NULL x_pmt=NULL
[24/04 05:39:07.024 main]: deleting pids on adapter 2, sid 0, pids=NULL
[24/04 05:39:07.024 main]: adding pids to adapter 2, sid 0, pids=0
[24/04 05:39:07.024 main]: Dumping pids table for adapter 2, number of unknown pids: 0
[24/04 05:39:07.024 main]: pid 0, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.024 main]: adding pids to adapter 2, sid 0, pids=326,406,3306
[24/04 05:39:07.024 main]: Dumping pids table for adapter 2, number of unknown pids: 0
[24/04 05:39:07.024 main]: pid 0, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.024 main]: pid 326, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.024 main]: pid 406, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.024 main]: pid 3306, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.024 main]: set_socket_thread: thread 7fae5bfff700 for sockets 8
[24/04 05:39:07.419 main]: tuning to 12399000(1120000) pol: h (2) sr:27500000 fec: delsys:dvbs2 mod:8psk rolloff: pilot:, ts clear=450040, ts pol=450305
[24/04 05:39:07.419 main]: Mutex init OK 0x561ee6e6bb00
[24/04 05:39:07.419 main]: Mutex init OK 0x561ee72ce7a8
[24/04 05:39:07.419 main]: new filter 0 added for adapter 2, pid 0, flags 20, mask_len 0, master_filter 0
[24/04 05:39:07.419 main]: Mutex init OK 0x561ee72cf818
[24/04 05:39:07.419 main]: new filter 1 added for adapter 2, pid 17, flags 20, mask_len 0, master_filter 1
[24/04 05:39:07.419 main]: Dumping pids table for adapter 2, number of unknown pids: 0
[24/04 05:39:07.419 main]: pid 0, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter 0, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 326, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 406, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 3306, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: AD 2 [demux 2 0], setting filter on PID 0 for fd 14
[24/04 05:39:07.419 main]: AD 2 [demux 2 0], setting filter on PID 326 for fd 15
[24/04 05:39:07.419 main]: AD 2 [demux 2 0], setting filter on PID 406 for fd 16
[24/04 05:39:07.419 main]: AD 2 [demux 2 0], setting filter on PID 3306 for fd 17
[24/04 05:39:07.419 main]: Reply (handle 7) [192.168.0.57:39484] content_len:0, sock 7
[24/04 05:39:07.419 main]: Read RTSP (sock 7, handle 7) [192.168.0.57:39484] sid 0, len: 96
[24/04 05:39:07.419 main]: detect_dvb_parameters (S)-> addpids=1,16,17
[24/04 05:39:07.419 main]: detect_dvb_parameters (E) -> src=-1, fe=-1, freq=-1, fec=-1, sr=-1, pol=-1, ro=-1, msys=-1, mtype=-1, plts=-1, bw=-1, inv=-1, pids=NULL - apids=1,16,17 - dpids=NULL x_pmt=NULL
[24/04 05:39:07.419 main]: Setup stream sid 0 parameters, sock_id 7, handle 7
[24/04 05:39:07.419 main]: copy_dvb_param start -> src=1, fe=1, freq=12399000, fec=9, sr=27500000, pol=2, ro=3, msys=6, mtype=9, plts=2, bw=8000000, inv=2, pids=NULL, apids=NULL, dpids=NULL x_pmt=NULL
[24/04 05:39:07.419 main]: copy_dvb_parameters -> src=1, fe=1, freq=12399000, fec=9 sr=27500000, pol=2, ro=3, msys=6, mtype=9, plts=2, bw=8000000, inv=2, pids=NULL, apids=1,16,17, dpids=NULL x_pmt=NULL
[24/04 05:39:07.419 main]: Play for stream sid 0, type 2, rsock 8, adapter 2, sock_id 7, rsock_id 9, handle 7
[24/04 05:39:07.419 main]: setting DVB parameters for adapter 2 - master_sid 0 sid 0 old f:12399000
[24/04 05:39:07.419 main]: copy_dvb_param start -> src=1, fe=1, freq=12399000, fec=9, sr=27500000, pol=2, ro=3, msys=6, mtype=9, plts=2, bw=8000000, inv=2, pids=0, apids=1,16,17, dpids=NULL x_pmt=NULL
[24/04 05:39:07.419 main]: copy_dvb_parameters -> src=1, fe=1, freq=12399000, fec=9 sr=27500000, pol=2, ro=3, msys=6, mtype=9, plts=2, bw=8000000, inv=2, pids=NULL, apids=1,16,17, dpids=NULL x_pmt=NULL
[24/04 05:39:07.419 main]: adding pids to adapter 2, sid 0, pids=1,16,17
[24/04 05:39:07.419 main]: Dumping pids table for adapter 2, number of unknown pids: 0
[24/04 05:39:07.419 main]: pid 0, fd 14, packets 0, d/c errs 0/0, flags 1, pmt -1, filter 0, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 326, fd 15, packets 0, d/c errs 0/0, flags 1, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 406, fd 16, packets 0, d/c errs 0/0, flags 1, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 3306, fd 17, packets 0, d/c errs 0/0, flags 1, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 1, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 16, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: pid 17, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.419 main]: not tuning for sid 0 (do_tune=0, master_sid=0)
[24/04 05:39:07.420 main]: Dumping pids table for adapter 2, number of unknown pids: 0
[24/04 05:39:07.420 main]: pid 0, fd 14, packets 0, d/c errs 0/0, flags 1, pmt -1, filter 0, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.420 main]: pid 326, fd 15, packets 0, d/c errs 0/0, flags 1, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.420 main]: pid 406, fd 16, packets 0, d/c errs 0/0, flags 1, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.420 main]: pid 3306, fd 17, packets 0, d/c errs 0/0, flags 1, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.420 main]: pid 1, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.420 main]: pid 16, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.420 main]: pid 17, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter 1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[24/04 05:39:07.420 main]: AD 2 [demux 2 0], setting filter on PID 1 for fd 18
[24/04 05:39:07.420 main]: AD 2 [demux 2 0], setting filter on PID 16 for fd 19
[24/04 05:39:07.420 main]: AD 2 [demux 2 0], setting filter on PID 17 for fd 20
[24/04 05:39:07.420 main]: Reply (handle 7) [192.168.0.57:39484] content_len:0, sock 7
[24/04 05:39:08.227 signal]: get_signal_new took 133 ms for adapter 2 handle 11 (status: 0, ber: 0, strength:202, snr: 0, force scan 0)
**[24/04 05:39:08.321 AD2]: no data sent for more than 1s sid: 0 for 192.168.0.57:45264
[24/04 05:39:08.321 AD2]: Start streaming for stream sid 0, len 12 to handle 8 => 192.168.0.57:45264
[24/04 05:39:09.422 AD2]: no data sent for more than 1s sid: 0 for 192.168.0.57:45264
Message repeated 6 times**
[24/04 05:39:17.088 main]: Read RTSP (sock 7, handle 7) [192.168.0.57:39484] sid 0, len: 84
[24/04 05:39:17.088 main]: closing stream sid 0
[24/04 05:39:17.088 main]: Closing RTP sock 9 handle 8
[24/04 05:39:17.088 main]: stream.c:333 get_sid returns NULL for s_id = 0

**Comment by 9000h** [comment]
ok IDLU-UWT110-CUO1O-32P I do have the same
do you have wideband LNB's ?

what is programmed in the switch, it can do different modes

-j 0:13-1400,1:23-1750,2:5-1155,3:14-1435 and -a is not needed
your settings look totally wrong to me as the slot/freq/pin you use are not listet for both switches you have

there is also a issue with the  power save mode for the Inverto switch in early models/firmware( the fixed is v0.25)

**Comment by Neolo** [comment]
Dude, I said that EVERY SINGLE tuner (either software on the PC or external like Dreambox) are TUNING FINE! Minisatip DOES NOT. Do you see a logic in here?

> your settings look totally wrong to me as the slot/freq/pin you use are not listet for both switches you have

They are not wrong, they are custom! This is why there is programmer exists, so you can customize it. But it's all doesn't matter, because read first sentence again. My inverto just arrived from the factory with hella fresh firmware, not even saying that completely unrelated brand Terra SRQ-540 is not being tuned by Minisatip the same way.

**Comment by MusterGit** [comment]
@Neolo 
This is not the first time that your tone is pretty harsh. You should be grateful that people are willing to help you.

**Comment by Neolo** [comment]
> @Neolo
> This is not the first time that your tone is pretty harsh. You should be grateful that people are willing to help you.

Well, sorry, people who don't bother to read the post entirely and just replying something completely nonsense is not just not helpful, but also making me look like a fool, this is annoying.

**Comment by 9000h** [comment]
so it should be (can not say if you channel for 1750 is right), but only slot 5 was set right so far as inverto start counting with slot 1 but you have to start with slot 0 in the minisatip config
so it should look like
-j 0:12-1400,1:23-1750,2:5-1155,3:13-1435 and -a is not needed

**Comment by Neolo** [comment]
> so it should be (can not say if you channel for 1750 is right), but only slot 5 was set right so far as inverto start counting with slot 1 but you have to start with slot 0 in the minisatip config
> so it should look like
> -j 0:12-1400,1:23-1750,2:5-1155,3:13-1435 and -a is not needed

I'm aware of that and I have tried that already, thanks. Frequency slot has to be tuned by another tuner first, before minisatip can read it.

**Comment by Neolo** [comment]
@9000h Why don't you try to make your switch Unicable 2 only (get rid of 1 version setup), set all pins on 0, all slots DYNAMIC and go ahead tune something over minisatip without prior use of any other tuning software. You'll be surprised.

**Comment by 9000h** [comment]
my setup works for minisatip, VDR and Panasonic TV's with the same switch and correct config

if you believe there is a issue with your switches then you have to ask the vendor
nobody did report such issue here and also not for satip-axe over the last 5 years

**Comment by Neolo** [comment]
> with the same switch and correct config

No comments... We are done with you, thanks.

![Screen Shot 2021-04-24 at 11 40 02 PM](https://user-images.githubusercontent.com/2836790/115988649-69755600-a556-11eb-892e-9f352874b4a1.png)

**Comment by 9000h** [comment]
@Neolo 
please do not post pictures without appropriate credit and link to the license

**Comment by Neolo** [comment]
> @Neolo
> please do not post pictures without appropriate credit and link to the license

Report me. In that case, you had no license from Inverto to screenshot their program and upload here. Smarty.

**Comment by catalinii** [comment]
@Neolo please be respectful towards people that are donating their time to help. If the advice does not work maybe someone else has anotger idea that can help.
Ultimately, this is the nature of free software, best effort.

**Comment by Neolo** [comment]
> @Neolo please be respectful towards people that are donating their time to help. If the advice does not work maybe someone else has anotger idea that can help.
> Ultimately, this is the nature of free software, best effort.

I’ve donated 5 days of my time of testing, and apparently there is a valuable result, which can add 1/4 to your wiki already. How about to  get some respect for that?

**Comment by catalinii** [comment]
I think I was respectful towards your time and contribution, but my point was for all of us to be respectful regardless if the advice worked or no.

**Comment by Neolo** [comment]
I set all related to minisatip jess slots in static mode, assigning needed transponders on the switch. This the only working solution and I advised it myself.Thanks.
_**Not**_ solved.

**Comment by catalinii** [comment]
If you have a working enigma2 receiver and minisatip the best way would be to compare the commands:
With enigma2 set the log to debug, start enigma2 tune on a channel, capture the log.
With minisatip run:
./minisatip -f -l http,dvb 
Capture the output and attach both.

Please make sure that when you tune to the same channel, both enigma2 and minisatip use the same unicable channel, freq, pin and so on

**Comment by 9000h** [comment]
--diseqc-timing is probably a issue when cheap DVB cards like TBS and bad drivers are in use, they need probably 40ms more time

**Comment by Neolo** [comment]
> --diseqc-timing is probably a issue when cheap DVB cards like TBS and bad drivers are in use, they need probably 40ms more time

Cheap? It's >$300 worth 6909 V2 model. Aint cheap.

**Comment by Neolo** [comment]
Dynamic mode for JESS is not working in any versions of minisatip. Why nobody can test this? Super easy.

**Comment by Jalle19** [comment]
I have an IDLU-32UL40-UNBOO-OPP LNBs and everything works just fine with a very standard `-j 0:0-1210,1:1-1420,2:2-1680,3:3-2040` configuration. Have you tried adding `-Z *:0` to keep the switch "active"? At least in my case this is needed, otherwise the first tune will fail. I've always chalked it up to the LNB, but maybe it's an issue with Unicable switches too :shrug:

**Comment by Neolo** [comment]
> 

issue is with multiswitch, not LNB. -z does nothing, because switch is always active. Minisatip just doesn't lock to a dynamic UB unless I set it statically. While Tvheadend and Cesbo Astra have no issues to work with dynamic mode UBs.

**Comment by Jalle19** [comment]
In what way is a Unicable capable multi-switch different from a Unicable LNB from a tuning/locking perspective?

**Comment by Neolo** [comment]
> In what way is a Unicable capable multi-switch different from a Unicable LNB from a tuning/locking perspective?

First of all this issue is about JESS mode with dynamic User Bands. Is this applicable to your LNB?

**Comment by Jalle19** [comment]
What do you mean by dynamic user bands? Isn't that the default?

**Comment by Neolo** [comment]
> What do you mean by dynamic user bands? Isn't that the default?

On my switch dyn mode is default and as I stated, minisatip is not locking to desired transponder in this mode.
Dynamic means that UB can tune to requested TP by request, while static mode is when I assign a TP to an UB on the switch and client (minisatip in this case) is not able to override that TP. So it works only when I assign it statically.

**Comment by Jalle19** [comment]
Hmm, I use dynamic mode (static mode needs a programmer) and I have no issues 🤷 Maybe this is something specific to your switch somehow?

**Comment by Neolo** [comment]
> Hmm, I use dynamic mode (static mode needs a programmer) and I have no issues 🤷 Maybe this is something specific to your switch somehow?

Yes, apparently. As I said, no other software has the issue but minisatip.

**Comment by Neolo** [comment]
Huh?

[17/02 08:37:55.917 main]: Detected adapter 2 handle 7 DVB Card Name: TurboSight TBS 6590 DVB-S/S2/S2X/T/T2/C/C2/ISDB-T +
[17/02 08:37:55.917 main]: Detected delivery system for adapter 2: dvbs [5]
[17/02 08:37:55.917 main]: Detected delivery system for adapter 2: dvbs2 [6]
[17/02 08:37:55.917 main]: Detected delivery system for adapter 2: dss [4]
[17/02 08:37:55.917 main]: Mutex init OK 0x55ddd55cbf98
[17/02 08:37:55.917 main]: sockets_add: handle 8 (type 5) returning socket sock 7 [:0] read: 0x55ddd51e3acc
[17/02 08:37:55.917 main]: set_socket_thread: thread 7f6527ab1700 for sockets 7
[17/02 08:37:55.917 AD2]: Starting select_and_execute on thread ID 7f6527ab1700, thread_name AD2
[17/02 08:37:59.069 main]: initializing CA, fd 9 type 2 flags 0x3, after 3152 ms
[17/02 08:37:59.069 main]: slotid: 0
[17/02 08:37:59.069 main]: tcid: 1
[17/02 08:37:59.069 main]: done opening adapter 2 delivery systems: dvbs dvbs2 **dss undefined**

**Comment by Jalle19** [comment]
Can you open a separate ticket for that?

**Comment by Neolo** [comment]
> Can you open a separate ticket for that?

Why is it suppose to be separate ticket? I believe it's related to the issue why minisatip isn't tuning my jess switch. Could be tuner involved.

**Comment by Jalle19** [comment]
It's not related to that, the code simply prints four delivery systems and most adapters don't support that many. I get `done opening adapter 3 delivery systems: dvbc dvbt undefined undefined` for example.

**Comment by Jalle19** [comment]
Closing this for now. Dynamic mode must be working, otherwise there would be tons of people commenting here. Not sure why this seems to only affect you but it could be related to your particulsr switch 🤷

**Comment by Neolo** [comment]
> Closing this for now. Dynamic mode must be working, otherwise there would be tons of people commenting here. Not sure why this seems to only affect you but it could be related to your particulsr switch 🤷

How can you close it if neither you or anyone here tested mentioned switches Inverto IDLU-UWT110-CUO1O-32P and Terra SRQ540 to prove it otherwise?

**Comment by Jalle19** [comment]
Because the title implies it's not working at all when it very much is. Maybe if the issue was limited to Unicable switches, although I don't know how those differ from LNBs.

**Comment by Jalle19** [comment]
It seems @9000h also said he has the same switch and it works for him?

**Comment by Neolo** [comment]
> Because the title implies it's not working at all when it very much is. Maybe if the issue was limited to Unicable switches, although I don't know how those differ from LNBs.

Name of the issue can be edited, as it's about JESS switches, and issue confirmed with two different brand of such switches. Test with a converter should not be a valid reason to just close the issue.

**Comment by Neolo** [comment]
> It seems @9000h also said he has the same switch and it works for him?

I don't see any evidence it's working for him, he did not provide his minisatip config for his switch and his screenshot looks factory default, not clear if he tuned Unicable 1 protocol only or how he tested it or if he tested it at all.

**Comment by Jalle19** [comment]
It would be nice if someone could confirm that this happens
